mutation_partition.cc

```diff
@@ -526,6 +541,36 @@ void mutation_partition::apply_weak(const schema& s, mutation_partition&& p, mut
     apply_monotonically(s, std::move(p), no_cache_tracker, app_stats);
 }
 
+future<> mutation_partition::apply_weak_gently(const schema& s, mutation_partition_view p,
+        const schema& p_schema, mutation_application_stats& app_stats) {
+    // FIXME: Optimize
+    mutation_partition p2(*this, copy_comparators_only{});
+    partition_builder b(p_schema, p2);
+    co_await p.accept_gently(p_schema, b);
+    apply_resume res;
+    while (apply_monotonically(s, std::move(p2), p_schema, app_stats, is_preemptible::yes, res) == stop_iteration::no) {
+        co_await coroutine::maybe_yield();
+    }
+}
+
+future<> mutation_partition::apply_weak_gently(const schema& s, const mutation_partition& p,
+        const schema& p_schema, mutation_application_stats& app_stats) {
+    // FIXME: Optimize
+    mutation_partition p2(s, p);
+    apply_resume res;
+    while (apply_monotonically(s, std::move(p2), p_schema, app_stats, is_preemptible::no, res) == stop_iteration::no) {
+        co_await coroutine::maybe_yield();
+    }
+}
+
+future<> mutation_partition::apply_weak_gently(const schema& s, mutation_partition&& p, mutation_application_stats& app_stats) {
+    mutation_partition p2(std::move(p));
+    apply_resume res;
+    while (apply_monotonically(s, std::move(p2), s, app_stats, is_preemptible::no, res) == stop_iteration::no) {
```

Comment:
Why not use `p` directly?

Did you mean is_preemptible::yes?

Reply:
> Why not use p directly?

Because it will disappear if the coroutine returns

> Did you mean is_preemptible::yes?

Yes, thanks (copy-pasta)

Reply:
> Because it will disappear if the coroutine returns

Right.